assets/src/features/room-page/utils/usePinning.tsx

```diff
@@ -1,15 +1,26 @@
 import { useState } from "react";
 
 export type PinningApi = {
-  pinnedTileId: string;
+  pinnedTileIds: string[];
   pin: (tileId: string) => void;
-  unpin: () => void;
+  unpin: (tileId: string) => void;
 };
 
 const usePinning = (): PinningApi => {
-  const [pinnedTileId, setPinnedTileId] = useState<string>("");
+  const [pinnedTileIds, setPinnedTileIds] = useState<string[]>([]);
 
-  return { pinnedTileId, pin: setPinnedTileId, unpin: () => setPinnedTileId("") };
+  const pin = (newTileId: string) => {
+    console.log(pinnedTileIds);
+    if (!pinnedTileIds.includes(newTileId)) setPinnedTileIds((oldPinnedTileIds) => [newTileId, ...oldPinnedTileIds]);
+    console.log(pinnedTileIds);
+  };
+
+  const unpin = (tileIdToRemove: string) => {
+    const idsWithoutGivenTile = pinnedTileIds.filter((tileId) => tileId !== tileIdToRemove);
+    setPinnedTileIds(idsWithoutGivenTile);
```

Comment:
It's safer to make this calculation inside `setPinnedTileIds`. There is most recent state

```ts
    setPinnedTileIds((prevState) => prevState.filter((tileId) => tileId !== tileIdToRemove));
```